RFC.md

```diff
@@ -341,6 +341,183 @@ const uploadedObjectReference = await uploadTask.result;
 
 Try out the new `storage` types here: https://www.typescriptlang.org/play#gist/292f4e24178bfca5881aa20961b930dc
 
-# `API` & `Datastore` Changes
+# `API` category Changes
+
+## First param is an object with named parameters
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```typescript
+const apiName = 'MyApiName';
+const path = '/path';
+const myInit = {
+  headers: {}, // OPTIONAL
+  response: true, // OPTIONAL (return the entire Axios response object instead of only response.data)
+  queryStringParameters: {
+    name: 'param' // OPTIONAL
+  }
+};
+
+API.get(apiName, path, myInit)
+  .then((response) => {
+    // Add your code here
+  })
+  .catch((error) => {
+    console.log(error.response);
+  });
+```
+
+```typescript
+API.get({
+    apiName: 'MyApi',
+    path: '/items',
+    authMode: 'AWS_IAM'
+}, myInit)
+  .then(result => {
+    // Add your code here
+  }).catch((error) => {
+    console.log(error.response);
+  });
+```
+
+## Adding TypeScript generics to request body and request response
+
+**Amplify v5 (`aws-amplify@5`)**
+
+Not possible to use generics for request body or response body
+
+
+**Proposed Amplify v6 (`aws-amplify@6`)**
+
+```typescript
+type MyApiResponse = { firstName: string, lastName: string };
+
+API.get<MyApiResponse>({
+    apiName: 'MyApi',
+    path: '/getName'
+}).then(result => {
+    console.log(`The name is ${result.body.firstName} ${result.body.lastName}`)
+});
+
+API.put<string, { data: Array<number> }>({
+    apiName: '',
+    path: '/',
+    authMode: "API_KEY"
+}, {
+    headers: {
+        "Content-type": "text/plain",
+    },
+    body: "this is my content"
+}).then(result => {
+    result.body.data.forEach(value => console.log(value));
+});
+```
+
+## GraphQL operations are splitted between query, mutation and subscription
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```typescript
+import { API } from "aws-amplify";
+import * as mutations from './graphql/mutations';
+import { GraphQLQuery, GraphQLSubscription } from '@aws-amplify/api';
+import { CreateTodoInput, CreateTodoMutation, OnCreateTodoSubscription } from './API';
+import * as subscriptions from './graphql/subscriptions';
+
```

Comment:
```suggestion
```

IMO we should omit imports from samples.

Reply:
Yeah, it was to make explicit the thing is annoying lol